### ISSUE TITLE
Warn about missing dollar sign near braces within double quotes

### ISSUE THREAD

**Issue by dangom** [koalaman/shellcheck#939]
#### Feature Request

- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

foo="bar"
echo "${foo}_{foo}"
```

#### Here's what shellcheck currently says:
$ shellcheck myscript
No issues detected!

#### Here's what I wanted or expected to see:

Similar to [SC1083,](https://github.com/koalaman/shellcheck/wiki/SC1083) but for braces within double quoted expressions.

$ shellcheck myscript
This { is literal. Check expression (missing $?)

**Comment by pkitszel** [comment]
Would be also useful for cases like:
`main "@"` # possibly `main "$@"` was intended

**Comment by cD1rtX3** [comment]
−1. While it *could* be unintentional, it most likely isn't. Also, there would be no way to silence this warning, since `"\{"` actually just treats the backslash as a literal backslash, not an escape character.
